src/inmemory.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * In-memory version of the `__aux` wrapper used in entity.js
+ */
+
+var _      = require('lodash');
+var crypto = require('crypto');
+
+// the in-memory data; stored globally as this is a better model for
+// Azure than storing each table as an instance or class property
+var tables = {};
+
+var InMemoryWrapper = function InMemoryWrapper(table) {
+  this.table = table;
+};
+
+/* Internal utilities */
+
+var entityEtag = function(entity) {
+  var sha1 = crypto.createHash('sha1');
+  // filter out metadata
+  var odata = /^odata\.|@odata/;
+  entity = _.omit(entity, function(v, k) { return odata.test(k); });
+  sha1.update(JSON.stringify(entity));
```

Comment:
Use `json-stable-stringify`, using `JSON.stringify()` will bite you badly one day :)